### ISSUE TITLE
Logging payload inconsistencies based on global library requirements

### ISSUE THREAD

**Issue by 0xSage** [googleapis/java-logging#344]
For internal Google devs: this is a mega issue tracker as a result of some cross language/platform [benchmarking work](https://docs.google.com/document/d/1yL0YXM5grqmqMuab-lKHvzP_MvnFfmK8ys24GDFIWwQ/edit#heading=h.p0ldqk1823yr). 

@codeowner, feel free to modify or discard suggestions based on language standards or not reproducible:

**Across all GCP platforms**
- [x] STDOUT logs via `System.out.printf(stdlog);` do not show up in Pantheon/log view. To reproduce, try [these logging quickstarts](https://user.git.corp.google.com/nicolezhu/log-benchmark/+/refs/heads/main/java/) in GKE or GAE
- [x] log entries missing resource auto-detection, e.g. `gce_instance`, `gae_app` as in Go/Node/Python
- [x] log entries missing trace & span ids
- [x] httprequest type logs missing protocol autodetection, i.e. httpRequest.protocol should default to `HTTP/1.1` based on requestMethod
- [ ] httprequest.NewBuilder.setRequestMethod (and other methods) should take in url in string format (in addition to the custom RequestMethod type. According to API docs

**Platform specific**
- [ ] GCE: Logging client library doesn't log in GCE (though it could be user configuration issue). To reproduce, try deploying this [GCE x logging quickstart](https://user.git.corp.google.com/nicolezhu/log-benchmark/+/refs/heads/main/java/gce/)
- [ ] GCF: execution_id not captured
- [ ] GCF: Logging client library doesn't log in GCF (known issue). I was able to reproduce this unresolved/closed issue #33 from before. To reproduce, try deploying this [boilerplate](https://user.git.corp.google.com/nicolezhu/log-benchmark/+/refs/heads/main/java/gcf/)
- [ ] GCF: A naming conflict arises when using logging library in Cloud Functions: 
```
com.google.cloud.logging.HttpRequest
com.google.cloud.functions.HttpRequest
```
Usage for the user become verbose. Similar to the python naming issue @Daniel-Sanche is looking at

**Comment by simonz130** [comment]
@minherz - this bug is great as it can be treated as older friction log and now we can compare the structured logging outcome to this feedback and see if there is any other gap

**Comment by minherz** [comment]
Marking it resolved. All mentioned issues except for samples were addressed in the recently released 3.6.0.
Samples will be updated in Q1'22.
